src/SprykerEco/Zed/Adyen/Business/Hook/Saver/MakePayment/CreditCardSaver.php

```diff
@@ -49,13 +52,20 @@ protected function updatePaymentAdyenTransfer(
      */
     protected function getPaymentStatus(?PaymentAdyenTransfer $paymentAdyenTransfer = null): string
     {
-        if (
-            $paymentAdyenTransfer
-            && $paymentAdyenTransfer->getResultCode() === $this->config->getOmsStatusRefused()
-        ) {
-                return $this->config->getOmsStatusRefused();
+        if ($paymentAdyenTransfer && $this->hasValidResultCode($paymentAdyenTransfer)) {
```

Comment:
There are 12 possible result codes from Adyen side.
https://docs.adyen.com/online-payments/payment-result-codes

Reply:
I'm not sure but I did next things:
https://github.com/spryker-eco/adyen/pull/36/files#diff-97d2b99877a1cb741dc71a7e82438c2fee825f2773e073c88af02f83183311c2


As I understand status from response of external adyen system and our OMSStatus it's not the same.
For example:
Adyen has status **Authorised** but our `OMS_STATUS_AUTHORIZED = 'authorized';`

So I match status **refused** with `OMS_STATUS_REFUSED`
and **authorised** with `OMS_STATUS_AUTHORIZED`.


Other situation wil be **OMS_STATUS_NEW**.

It seems like will be working.
What do you think?

Reply:
If I am not mistaken the following result codes from Adyen also means refused: `Cancelled` and `Error`.

Reply:
Yes, agree.
Mathed `Cancelled` and `Error` as OMS_STATUS_REFUSED